### PR TITLE
Replace deprecated Checkbox react component with web component

### PR DESCRIPTION
**Pull request by jamigibbs**
## Summary

This PR replaces the deprecated design system `Checkbox` React component with `va-checkbox` or `va-privacy-policy` (where relevant).

## Related issue(s)

- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1920

## Testing done
Locally. Tests passing and previewed visually in Cypress.

## Screenshots

**yellow-ribbon**

Before

yellow-ribbon-before

After

yellow-ribbon-after

**claim status AskVAPage**

Screenshot 2023-09-27 at 4 24 50 PM

**claim status**

Before

claim-status-after

After

claim-status-before

**covid 19 vaccine**

Before

covide-19-vaccine-before

After

Screenshot 2023-09-29 at 11 48 21 AM

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/yellow-ribbon/containers/SearchForm/index.unit.spec.jsx
Comment (jamigibbs) [inline src/applications/yellow-ribbon/containers/SearchForm/index.unit.spec.jsx:19]: There are only 2 inputs here so I'm not sure why this was checking for 4. Test was failing otherwise.

Screenshot 2023-09-27 at 3 31 35 PM

File: src/applications/yellow-ribbon/containers/SearchForm/index.jsx
Comment (jamigibbs) [inline src/applications/yellow-ribbon/containers/SearchForm/index.jsx:208]: The `onVaChange` event details is returned as `event.detail.checked` (boolean) but the implementation does not seem to want to know true/false (just the string "contributionAmount") so I have kept that the same.

**Review by powellkerry — APPROVED** [review]

File: src/applications/enrollment-verification/components/PrivacyAgreement.jsx
Comment (jamigibbs) [inline src/applications/enrollment-verification/components/PrivacyAgreement.jsx:13]: I have removed this entire file because it does not appear to be used any longer based on this PR: https://github.com/department-of-veterans-affairs/vets-website/pull/22348/files

If this is still needed/wanted, use `va-privacy-agreement` component instead: https://design.va.gov/storybook/?path=/docs/components-va-privacy-agreement--default

**Comment by randimays** [comment]
Re: Yellow Ribbon tool (this is the only app in this PR that we own): the filter is not working properly.

Here's a screenshot from production. I've clicked the second checkbox and hit "Search." Notice the checkbox remains checked.

Screenshot 2023-10-02 at 11 53 18 AM

<br/>
<br/>
<br/>
<br/>

Here are screenshots from the review instance. I've clicked the second checkbox and hit "Search." Notice the checkbox does not remain checked after clicking "Search," and also the results are not filtered properly. The search should only be displaying results for schools that provide funding to all eligible students, but the second screenshot (in the list of results) only covers 20 students.

Screenshot 2023-10-02 at 11 53 30 AM

Screenshot 2023-10-02 at 11 53 34 AM

**Comment by jamigibbs** [comment]
> Re: Yellow Ribbon tool (this is the only app in this PR that we own): the filter is not working properly.

That's a good callout (and that you for testing it!). When I tested it, I don't think the original implementation was working either. For example:

https://www.va.gov/education/yellow-ribbon-participating-schools/?name=University&state=IL

So I didn't think it not filtering was related to the web component. Could this be something your team can add to their backlog to investigate as a defect with the application itself (unrelated to changing it to the web component)?

**Comment by randimays** [comment]
@jamigibbs I tested https://www.va.gov/education/yellow-ribbon-participating-schools/?name=University&state=IL and both filters are working properly for me. What did you observe was not working?

**Comment by jamigibbs** [comment]
@randimays Disregard that last comment. I was thinking the checkbox would work on-click. 🤦🏻‍♀️ Let me do a little more poking around here with it.

**Comment by randimays** [comment]
@jamigibbs It seems like the checkbox _**should**_ work on click. I was confused by that too.

**Comment by jamigibbs** [comment]
@randimays Thanks again for the extra set of eyes. It looks like everything is filtering again after [updating the callback](https://github.com/department-of-veterans-affairs/vets-website/pull/25893/commits/4e45cea5229446fc0b5efbecf2420acb9f7e2c1e). 

I'd appreciate if you could double check it though 🙏: http://2ffa80215923e762336b6fd65ce3837b.review.vetsgov-internal/education/yellow-ribbon-participating-schools/?name=University&state=IL

File: src/applications/coronavirus-vaccination-expansion/config/PreSubmitInfo.js
Comment (jamigibbs) [inline src/applications/coronavirus-vaccination-expansion/config/PreSubmitInfo.js:50]: We have updated this link because the previous one does not work:

https://www.va.gov/vhapublications/ViewPublication.asp?pub_ID=1090

**Review by randimays — APPROVED** [review]
LGTM thanks @jamigibbs!

**Comment by jamigibbs** [comment]
Hello to these reviewing teams 👋  We are trying to merge this PR but we are blocked by your reviews. Can you please review the files that you are codeowners for?

@department-of-veterans-affairs/benefits-management-tools-fe 
@department-of-veterans-affairs/govcio-vfep-codereviewers 
@department-of-veterans-affairs/my-education-benefits
@department-of-veterans-affairs/va-cto-health-products
@department-of-veterans-affairs/vsa-debt-frontend

**Review by jazebnad — APPROVED** [review]

**Review by it-harrison — APPROVED** [review]
lgtm

**Review by Dr-Pongo — APPROVED** [review]
LGTM! 🚀

**Review by acrollet — APPROVED** [review]
approved for coronavirus-vaccination-expansion, ty for doing this work!

**Review by GovNapoleon — APPROVED** [review]

File: src/applications/edu-benefits/containers/PreSubmitInfo.jsx
Comment (jamigibbs) [inline src/applications/edu-benefits/containers/PreSubmitInfo.jsx:19]: va-privacy-agreement on storybook for reference: https://design.va.gov/storybook/?path=/docs/components-va-privacy-agreement--default

File: src/applications/claims-status/components/AddFilesForm.jsx
Comment (jerekshoe) [inline src/applications/claims-status/components/AddFilesForm.jsx:292]: There are 2 pleases here and in the div below

Comment (jamigibbs) [reply src/applications/claims-status/components/AddFilesForm.jsx:292]: Hi @jerekshoe Thanks for taking a look. I'm not sure I understand your comment here though. Is there something that needs to be updated?

Comment (jamigibbs) [reply src/applications/claims-status/components/AddFilesForm.jsx:292]: Oh, I see! There's a typo "**please please** visit our How to File a Claim page."

It looks like this was in the original code so I didn't notice but I'll get it cleaned up now.

**Comment by jamigibbs** [comment]
Hi @rsmithadhoc 👋 I was wondering if you could weigh in on the best way to handle the update to the [checkbox in the Claim Status add file](https://github.com/department-of-veterans-affairs/vets-website/blob/d9c0d9c7b1d525585020906fb16f2a5b8f6fe979/src/applications/claims-status/components/AddFilesForm.jsx#L286C1-L299C15) app component.

Previously, the React component checkbox included a link to "How to File a Claim" in label like this:

Screenshot 2023-10-04 at 3 36 32 PM

But we can not pass markup in the checkbox label with `va-checkbox`. I think there are a couple of different ways we can still provide that context though for visually impaired users.

1. Use the `message-aria-describedby` prop to pass additional content/message so that the user is aware that there is additional sited context to the checkbox (that is what this PR is currently doing). Something like this:

Screenshot 2023-10-04 at 3 21 48 PM

2. Use the [`description` slot to add JSX markup](https://design.va.gov/storybook/?path=/docs/components-va-checkbox--with-description-jsx) above the checkbox. That will look like this though (it changes the order of the text):

Screenshot 2023-10-04 at 3 50 00 PM

3. Maybe a combination of both?

---

Cypress test for previewing locally: `/applications/claims-status/tests/e2e/02.claim-files.cypress.spec.js`
Web Component Reference: https://design.va.gov/storybook/?path=/docs/components-va-checkbox--with-description-jsx

cc @jerekshoe

File: src/applications/claims-status/tests/components/AskVAPage.unit.spec.jsx
Comment (jamigibbs) [inline src/applications/claims-status/tests/components/AskVAPage.unit.spec.jsx:33]: The `SkinDeep.shallowRender` utility does not work with web components. An alternate approach we have had some success with is with `ReactTestUtils.renderIntoDocument` but I wasn't able to get that funcionioning entirely in this file for some reason.

**Comment by rsmithadhoc** [comment]
> Hi @rsmithadhoc 👋 I was wondering if you could weigh in on the best way to handle the update to the [checkbox in the Claim Status add file](https://github.com/department-of-veterans-affairs/vets-website/blob/d9c0d9c7b1d525585020906fb16f2a5b8f6fe979/src/applications/claims-status/components/AddFilesForm.jsx#L286C1-L299C15) app component.
> 
> Previously, the React component checkbox included a link to "How to File a Claim" in label like this:
> 
> Screenshot 2023-10-04 at 3 36 32 PM
> 
> But we can not pass markup in the checkbox label with `va-checkbox`. I think there are a couple of different ways we can still provide that context though for visually impaired users.
> 
>     1. Use the `message-aria-describedby` prop to pass additional content/message so that the user is aware that there is additional sited context to the checkbox (that is what this PR is currently doing). Something like this:
> 
> 
> Screenshot 2023-10-04 at 3 21 48 PM
> 
>     2. Use the [`description` slot to add JSX markup](https://design.va.gov/storybook/?path=/docs/components-va-checkbox--with-description-jsx) above the checkbox. That will look like this though (it changes the order of the text):
> 
> 
> Screenshot 2023-10-04 at 3 50 00 PM
> 
>     3. Maybe a combination of both?
> 
> 
> Cypress test for previewing locally: `/applications/claims-status/tests/e2e/02.claim-files.cypress.spec.js` Web Component Reference: https://design.va.gov/storybook/?path=/docs/components-va-checkbox--with-description-jsx
> 
> cc @jerekshoe

I think option #1 is the way to go.

**Review by jerekshoe — APPROVED** [review]

**Pull request merged**
